File: src/game/tools/rpc/output/wrap/PlayerWrap.go
package wraps

import (
	rpc "game/common/rpc/simple"
)

type PlayerWrap struct {
	
	UID int64 // 游戏的uid
	GUID string // 平台唯一ID,游客登录的话为设备ID
	PID string // 登录平台ID
	UserName string // 用户名（唯一，初始和UID一致）
	LoginTime int64 // 登录时间戳（秒）
	OnlineTime int64 // 总在线时长（秒）
	CreateTime int64 // 创角时间戳（秒）
	ServerTime int64 // 服务器时间（秒）
	OnBattleIdx int32 // 上阵索引
	Items []*ItemDataWrap // 道具列表
	Icon int32 // 头像
	TutorialMask int64 // 新手引导
	rpc.Wrapper
}

func (w *PlayerWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.UID = pck.PopInt64(); 
	w.GUID = pck.PopString(); 
	w.PID = pck.PopString(); 
	w.UserName = pck.PopString(); 
	w.LoginTime = pck.PopInt64(); 
	w.OnlineTime = pck.PopInt64(); 
	w.CreateTime = pck.PopInt64(); 
	w.ServerTime = pck.PopInt64(); 
	w.OnBattleIdx = pck.PopInt32(); 
	w.Items = make([]*ItemDataWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.Items); i++ {
		w.Items[i] = new(ItemDataWrap)
		w.Items[i].Decode(pck)
		
	}
	
	w.Icon = pck.PopInt32(); 
	w.TutorialMask = pck.PopInt64(); 
	return w
}

func (w *PlayerWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt64(w.UID) 
	pck.PutString(w.GUID) 
	pck.PutString(w.PID) 
	pck.PutString(w.UserName) 
	pck.PutInt64(w.LoginTime) 
	pck.PutInt64(w.OnlineTime) 
	pck.PutInt64(w.CreateTime) 
	pck.PutInt64(w.ServerTime) 
	pck.PutInt32(w.OnBattleIdx) 
	if w.Items == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.Items)))
    	for i := 0; i < len(w.Items); i++ {
    		w.Items[i].Encode(pck);
			
    	}
    }
	
	pck.PutInt32(w.Icon) 
	pck.PutInt64(w.TutorialMask) 
}

File: src/game/tools/rpc/output/wrap/PlayerWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class PlayerWrap : IWrapper
	{	
		public long UID; // 游戏的uid
		public string GUID; // 平台唯一ID,游客登录的话为设备ID
		public string PID; // 登录平台ID
		public string UserName; // 用户名（唯一，初始和UID一致）
		public long LoginTime; // 登录时间戳（秒）
		public long OnlineTime; // 总在线时长（秒）
		public long CreateTime; // 创角时间戳（秒）
		public long ServerTime; // 服务器时间（秒）
		public int OnBattleIdx; // 上阵索引
		public ItemDataWrap[] Items; // 道具列表
		public int Icon; // 头像
		public long TutorialMask; // 新手引导
		
		public void Decode(Packet pck)
		{	
			UID = pck.GetLong(); 
			GUID = pck.GetString(); 
			PID = pck.GetString(); 
			UserName = pck.GetString(); 
			LoginTime = pck.GetLong(); 
			OnlineTime = pck.GetLong(); 
			CreateTime = pck.GetLong(); 
			ServerTime = pck.GetLong(); 
			OnBattleIdx = pck.GetInt(); 
			Items = new ItemDataWrap[pck.GetShort()];
			for (int i = 0; i < Items.Length; i++)
			{
				Items[i] = new ItemDataWrap();
				Items[i].Decode(pck);
				
			}
			
			Icon = pck.GetInt(); 
			TutorialMask = pck.GetLong(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutLong(UID); 
        	pck.PutString(GUID); 
        	pck.PutString(PID); 
        	pck.PutString(UserName); 
        	pck.PutLong(LoginTime); 
        	pck.PutLong(OnlineTime); 
        	pck.PutLong(CreateTime); 
        	pck.PutLong(ServerTime); 
        	pck.PutInt(OnBattleIdx); 
        	if (Items == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)Items.Length);
	        	for(int i = 0; i < Items.Length; i++)
	        	{
	        		Items[i].Encode(pck);
					
	        	}
	        }
        	
        	pck.PutInt(Icon); 
        	pck.PutLong(TutorialMask); 
        }
	}
}